WcaOnRails/app/models/competition.rb

```diff
@@ -427,16 +439,24 @@ def dangerously_close_to?(c)
     self.kilometers_to(c) <= NEARBY_DISTANCE_KM_DANGER && days_until.abs < NEARBY_DAYS_DANGER
   end
 
+  def results_uploaded?
+    Result.exists?(competitionId: self.id)
+  end
+
   def user_can_view?(user)
     self.showAtAll || (user && user.can_manage_competition?(self))
   end
 
+  def in_progress?
+    start_date <= Date.today && end_date >= Date.today
```

Comment:
Three equal signs! `===`

It's a somewhat strange operator which doesn't have much to do with actual equality (it's not even commutative). Its meaning is all over the place depending on which object you call it. It's mainly used for desugaring `case when` expressions, e.g.

```ruby
case age
when 2..4
  "young"
when 5..100
  "old"
else
  "dead"
end
```

is equivalent to

```ruby
if (2..4) === age
  "young"
# ...
```